pkg/sql/logictest/testdata/logic_test/system

```diff
@@ -376,42 +375,77 @@ GRANT GRANT, SELECT ON system.namespace TO testuser
+statement error user root must have exactly {GRANT, SELECT} privileges on system object with ID=.*
 GRANT ALL ON DATABASE system TO root
 
-statement error user root must have exactly {GRANT, SELECT} privileges on this system object
+statement error user root must have exactly {GRANT, SELECT} privileges on system object with ID=.*
 GRANT DELETE, INSERT ON DATABASE system TO root
 
-statement error user root must have exactly {GRANT, SELECT} privileges on this system object
+statement error user root must have exactly {GRANT, SELECT} privileges on system object with ID=.*
 GRANT ALL ON system.namespace TO root
 
-statement error user root must have exactly {GRANT, SELECT} privileges on this system object
+statement error user root must have exactly {GRANT, SELECT} privileges on system object with ID=.*
 GRANT DELETE, INSERT ON system.descriptor TO root
 
-statement error user root must have exactly {GRANT, SELECT} privileges on this system object
+statement error user root must have exactly {GRANT, SELECT} privileges on system object with ID=.*
 GRANT ALL ON system.descriptor TO root
 
-statement error user root must have exactly {GRANT, SELECT} privileges on this system object
+statement error user root must have exactly {GRANT, SELECT} privileges on system object with ID=.*
 REVOKE GRANT ON DATABASE system FROM root
 
-statement error user root must have exactly {GRANT, SELECT} privileges on this system object
+statement error user root must have exactly {GRANT, SELECT} privileges on system object with ID=.*
 REVOKE GRANT ON system.namespace FROM root
 
 statement error user root does not have privileges
 REVOKE ALL ON system.namespace FROM root
 
+statement error user admin must have exactly {GRANT, SELECT} privileges on system object with ID=.*
+GRANT ALL ON DATABASE system TO admin
+
+statement error user admin must have exactly {GRANT, SELECT} privileges on system object with ID=.*
+GRANT DELETE, INSERT ON DATABASE system TO admin
+
+statement error user admin must have exactly {GRANT, SELECT} privileges on system object with ID=.*
+GRANT ALL ON system.namespace TO admin
+
+statement error user admin must have exactly {GRANT, SELECT} privileges on system object with ID=.*
+GRANT DELETE, INSERT ON system.descriptor TO admin
+
+statement error user admin must have exactly {GRANT, SELECT} privileges on system object with ID=.*
+GRANT ALL ON system.descriptor TO admin
+
+statement error user admin must have exactly {GRANT, SELECT} privileges on system object with ID=.*
+REVOKE GRANT ON DATABASE system FROM admin
+
+statement error user admin must have exactly {GRANT, SELECT} privileges on system object with ID=.*
+REVOKE GRANT ON system.namespace FROM admin
+
+statement error user admin does not have privileges
+REVOKE ALL ON system.namespace FROM admin
+
 # Some tables (we test system.lease here) allow multiple privilege sets for
-# backwards compatibility, but still enforce that no user has more privileges
```

Comment:
this was an oversight on my part when reviewing the changes to this code before (when "allowedPrivileges" became a list). It was never meant to check that "users have no more privileges than root", it was supposed to be "users have allowed privileges". The confusion came from the fact that "root" had exactly the allowed privileges, so it was a shortcut to not have to convert types again (from privilege kind to privilege bits).